### PR TITLE
Proxy: Socket file 'other' permission changes to support ACRN Hypervisor

### PR DESCRIPTION
**Pull request by vijaydhanraj**
When binding the socket with the file path, the socket file permission
is changed to 755. The reason being the following,

Bind system call sets the socket file permission(mode) based on,
socket FD permission and umask bits set. Socket FD by default is
set to 777 and umask by default is set to 022. This results in 755.
ACRN doesn't plan to have an API to update the umask so added
the workaround in kata proxy, where if the ‘other’ permission
is not disabled, then kata proxy can override and disable it.

Fixes: #191

Signed-off-by: Vijay Dhanraj <vijay.dhanraj@intel.com>

### CODE REVIEW

**Comment by vijaydhanraj** [comment]
Related to: https://github.com/kata-containers/runtime/pull/1779

File: proxy.go [hunk 1]
```diff
@@ -239,7 +239,17 @@ func printAgentLogs(sock string) error {
...
 	if other != 0 {
-		return fmt.Errorf("All socket permissions for 'other' should be disabled, got %3.3o", other)
+		logger().Infof("All socket permissions for 'other' should be disabled, got %4.4o", int(fileInfo.Mode().Perm()))
+
+		// Setting permissions socket for "other" to 0.
+		otherMask := 0770
+		other := int(fileInfo.Mode().Perm()) & otherMask
+		err := os.Chmod(agentLogsAddr, os.FileMode(other))
+		if err != nil {
+			logger().Infof("Cannot change socket permissions for 'other' %d", err)
```
Comment (devimc) [inline proxy.go]: use `WithField` and `Error`

```go
logger().WithField("error", err).Error("Cannot change socket permissions for 'other'")
```

File: proxy.go [hunk 2]
```diff
@@ -239,7 +239,17 @@ func printAgentLogs(sock string) error {
...
+		} else {
+			logger().Infof("Chaning socket permissions to %4.4o", other)
```
Comment (devimc) [inline proxy.go]: please move this log before `os.Chmod`  and use `WithField` instead of `%4.4o`

File: proxy.go [hunk 3]
```diff
@@ -239,7 +239,17 @@ func printAgentLogs(sock string) error {
 	otherMask := 0007
 	other := int(fileInfo.Mode().Perm()) & otherMask
 	if other != 0 {
-		return fmt.Errorf("All socket permissions for 'other' should be disabled, got %3.3o", other)
+		logger().Infof("All socket permissions for 'other' should be disabled, got %4.4o", int(fileInfo.Mode().Perm()))
```
Comment (devimc) [inline proxy.go]: use `WithField` instead of `%4.4o`

File: proxy.go (hunk 1 above)
Comment (vijaydhanraj) [reply proxy.go]: done.  used WithError instead of WithField as discussed.

File: proxy.go (hunk 2 above)
Comment (vijaydhanraj) [reply proxy.go]: done

File: proxy.go (hunk 3 above)
Comment (vijaydhanraj) [reply proxy.go]: done

File: proxy.go [hunk 4]
```diff
@@ -239,7 +239,18 @@ func printAgentLogs(sock string) error {
 	otherMask := 0007
 	other := int(fileInfo.Mode().Perm()) & otherMask
 	if other != 0 {
-		return fmt.Errorf("All socket permissions for 'other' should be disabled, got %3.3o", other)
+		logger().WithField("socket permissions", fmt.Sprintf("%4.4o", fileInfo.Mode().Perm())).
+			Infof("All socket permissions for 'other' should be disabled")
```
Comment (jodh-intel) [inline proxy.go]: Nit: This can now be `Info()` as you don't have any formats to expand in the string.

File: proxy.go [hunk 5]
```diff
@@ -239,7 +239,18 @@ func printAgentLogs(sock string) error {
...
+
+		// Setting permissions socket for "other" to 0.
+		otherMask := 0770
```
Comment (jodh-intel) [inline proxy.go]: You could make this `const`

File: proxy.go [hunk 6]
```diff
@@ -239,7 +239,18 @@ func printAgentLogs(sock string) error {
...
+		other := int(fileInfo.Mode().Perm()) & otherMask
+		logger().WithField("other", fmt.Sprintf("%4.4o", other)).Infof("Chaning socket permissions of 'other'")
```
Comment (jodh-intel) [inline proxy.go]: Nits:

- typo: "Changing".
- You can also use `Info()` here rather than `Infof()`.

File: proxy.go (hunk 4 above)
Comment (vijaydhanraj) [reply proxy.go]: done.

File: proxy.go (hunk 5 above)
Comment (vijaydhanraj) [reply proxy.go]: This is actually not a declaration. I was trying to reassign a new value to the previously declared variable. 
PS: I have now changed this from a declaration to an assignment.

File: proxy.go (hunk 6 above)
Comment (vijaydhanraj) [reply proxy.go]: done.

**Comment by vijaydhanraj** [comment]
@gwhaley: Can you please help me reproduce the PR-check failure?

**Comment by grahamwhaley** [comment]
Hi @vijaydhanraj Looks like we need to re-trigger the CIs as you updated the PR, so first we:
/test

and then, it looks like the PR-check CI had some sort of system failure, so let's re-trigger that as well:
/zuul-recheck

and then we see how that goes in a couple of hours...

File: proxy.go (hunk 5 above)
Comment (jodh-intel) [reply proxy.go]: Good point! :smile:

**Comment by amshinde** [comment]
@grahamwhaley Looks like the zuul checks are still failing here, can you take a look? I dont see any logs on the build site.

**Comment by grahamwhaley** [comment]
/zuul-recheck
Oh Zuul zuul.... sigh, so, even though I'm sure we've had the PR-check working (otherwise we could not have merged anything), the logs shown on this link starting from 21st March all show one of NODE_FAILURE/FAILURE/RETRY_LIMIT ?
Maybe the logs only show jobs that fail?? I dunno
https://zuul.opendev.org/t/kata-containers/builds?pipeline=PR-check

Anyway, it looks like this PR (#193 - that is the clue to finding the relevant items for this PR in the log/job list on the above link) shows we hit another NODE_FAILURE. One more time, with gusto then....
/zuul-recheck
(and slapping one a the front of the comment, just in case as well)

**Comment by grahamwhaley** [comment]
Doh, @vijaydhanraj @amshinde - heh, that is the PR-check we are looking at - that is the WIP Zuul CI check we have that *does not currently pass* ( see #154 ) - so, don't expect that to pass. The other two Zuul CIs have passed (WIP and SOB) - so, we are good to push the merge maybe...

**Comment by vijaydhanraj** [comment]
Any other comments on this patch?

**Comment by jodh-intel** [comment]
ok - ARM CI failure is known, so let's just get this merged...

**Comment by jodh-intel** [comment]
... which we cannot do as the branch has conflicts. Could you update please @vijaydhanraj ?

**Comment by vijaydhanraj** [comment]
I didn't close the issue, just pushed my changes after rebase. Please let me know anything else is needed.

**Comment by amshinde** [comment]
@jodh-intel @vijaydhanraj Looks like this got merged manually vs the github UI.
Weird as I saw this PR open today morning.
@jodh-intel Any idea about this?

**Comment by jodh-intel** [comment]
@amshinde - I tried to merge from the UI, got the green "confirm merge" button and clicked it, but then got an error saying the branch was conflicted. However, I checked by grabbing a copy of the PR locally and it merged correctly so this may have been a "github gremlin" :alien: I left the PR in an open state so not sure what happened subsequently.

**Pull request closed without merging**